app.js

```diff
@@ -38,6 +48,97 @@ app.post('/vegetables', helpers.validate, (req, res, next) => {
   res.status(201).json(vegetable)
 })
 
+app.delete('/vegetables/:id', (req, res, next) => {
+  const { vegetables } = data
+  const { id } = req.params
+  const vegetable = vegetables.find(veggie => veggie.id === id)
+
+  for(let i = 0; i < data.vegetables.length; i++) {
+    if (vegetables[i].id === id) {
+      vegetables.splice(i, 1);
+      const message = `${vegetable.name} with ID ${id} has been deleted!`
+      next({ status: 200, message })
+    }
+  }
+
+  const message = `Could not find vegetable with ID of ${id}`
+  next({ status: 404, message })
+})
+
+app.put('/vegetables/:id', (req, res, next) => {
+  const { vegetables } = data
+  const { id } = req.params
+  const vegetable = vegetables.find(veggie => veggie.id === id)
+
+  let veggie;
+  for(let i = 0; i < data.vegetables.length; i++) {
+    if (vegetables[i].id === id) {
+      veggie.name = req.body.name
```

Comment:
I am unable to get through this route. I think the problem is that you've declared `veggie` up above, which means its value is set to `undefined`. You are then trying to assign a property of `.name` to it. This will result in a failure.